sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -411,15 +449,19 @@ private void processDeliveredMessage(Delivery delivery) {
             return;
         }
 
-        if (outcome instanceof Accepted) {
+        if (outcome instanceof Accepted
```

Comment:
I meant, line 452 onwards to line 463. The only time `Mono<DeliveryState> send` returns is in a success (Accepted/Declared) state.
    1. Why return DeliveryState if you already process it rather than `Mono<Void>`?
    2. It could be decoupled by returning the `DeliveryState` as-is in `processDeliveredMessage` if
    you have an associated `DeliveryState` (ie. TransactionState) that was passed.
        1. In this case, we could decouple any transaction handling from ReactorSender.

```java
class ReactorSender {
    Mono<DeliveryState> send(byte[] bytes, int arrayOffset, int messageFormat, DeliveryState deliveryState) {
        return validateEndpoint()
            .then(Mono.create(sink -> sendWork(new RetriableWorkItem(bytes,
                arrayOffset, messageFormat, sink, timeout, deliveryState)))
            );
    }

    private void processDeliveredMessage(Delivery delivery) {
        final DeliveryState outcome = delivery.getRemoteState();
        final String deliveryTag = new String(delivery.getTag(), UTF_8);

        logger.verbose("entityPath[{}], linkName[{}], deliveryTag[{}]: process delivered message",
            entityPath, getLinkName(), deliveryTag);

        final RetriableWorkItem workItem = pendingSendsMap.remove(deliveryTag);

        if (workItem == null) {
            logger.verbose("clientId[{}]. path[{}], linkName[{}], delivery[{}] - mismatch (or send timed out)",
                handler.getConnectionId(), entityPath, getLinkName(), deliveryTag);
            return;
        } else if (workItem.updatedDeliveryState()) {
            workItem.success(outcome);
            return;
        }

        // All the original content from line 444 without any transaction stuff.
    }
}

class RetriableWorkItem {
    private final DeliveryState deliveryState;

    boolean updatedDeliveryState() {
        return deliveryState != null;
    }
}

class TransactionCoordinator {
    Mono<Void> completeTransaction(AmqpTransaction transaction, boolean isCommit) {
        final int payloadSize = messageSerializer.getSize(message);
        final int allocationSize = payloadSize + MAX_AMQP_HEADER_SIZE_BYTES;

        final byte[] bytes = new byte[allocationSize];
        final int encodedSize = message.encode(bytes, 0, allocationSize);
        final DeliveryState deliveryState = new TransactionalState();
        deliveryState.setTxnId(new Binary(transaction.getTransactionId().array()));

        return sendLink.send(bytes, encodedSize, DeliveryImpl.DEFAULT_MESSAGE_FORMAT,deliveryState)
            .handle((outcome, MonoSink<Void> sink) -> {
                if (!(outcome instanceof TransactionalState)) {
                    sink.error(new IllegalArgumentException("Expected a TransactionalState, received: " + outcome));
                    return;
                }

                final TransactionalState state = (TransactionalState) outcome;
                final DeliveryStateType stateType = state.getType();
                switch (stateType) {
                    case Accepted:
                        break;
                    case Rejected:
                        break;
                    case Declared:
                        break;
                    default:
                }
            });
    }
}